edge-modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -27,7 +27,7 @@ public abstract class DirectMethodSenderBase : IDisposable
 
         public abstract void Dispose();
 
-        public async Task<HttpStatusCode> InvokeDirectMethodAsync(CancellationTokenSource cts)
+        public async Task<Tuple<HttpStatusCode, long>> InvokeDirectMethodAsync(CancellationTokenSource cts)
```

Comment:
I think you need to pass tracking Id and directMethodCount when calling this.InvokeDirectMethodAsync; therefore when DM receiver received your call, it can extract trackingId and directMethodCount; and construct TestOperationResult to report to TRC.

Reply:
It is done below. Line:58-68

Reply:
The InvokeDeviceMethodAsync(string deviceId, string targetModuleId, CancellationToken none) signature only has device Id, target module id; but no tracking id and direct method count.  How DM receiver can get these 2 numbers?

Reply:
I see, I misread. I thought you meant the TRC, not the DMReceiver .
I think that should be a separate PR to have DMReceiver takes the Connectivity params.

Reply:
Please add todo here if you plan to do it in another PR.

Reply:
The task: https://msazure.visualstudio.com/One/_workitems/edit/5954482 was created to address this issue.